component/http/handler.go

```diff
@@ -106,8 +108,9 @@ func getSingleHeaderEncoding(header string) (string, encoding.DecodeFunc, encodi
 	var dec encoding.DecodeFunc
 	var ct string
 
-	switch header {
-	case "*/*", json.Type, json.TypeCharset:
+	parts := strings.SplitN(header, ";", 2)
```

Comment:
we might have a problem here that could result in a panic. if the header is empty the above returns and empty slice and you access the first element.
check  https://go.dev/play/p/B6Q0QZIBXAN

Reply:
It looks like it returns an slice with length 1 where the item is an empty string
https://go.dev/play/p/Fw8rAfV08We